app/controllers/kernel_configs_controller.rb

```diff
@@ -21,7 +21,7 @@ def edit
 
   # POST /kernel_configs or /kernel_configs.json
   def create
-    @kernel_config = KernelConfig.new(kernel_config_params)
+    @kernel_config = KernelConfig.new(user_id: current_user.id, config_url: params[:config_url])
```

Comment:
We can take advantage of the association to build a `KernelConfig` for a user directly. This would then also let us use our filtered parameter method `kernel_config_params`.
```suggestion
    @kernel_config = current_user.kernel_configs.build(kernel_config_params)
```